### DISCUSSION TITLE
Attachments - Different scenarios in BudiBase

### DISCUSSION THREAD

**Discussion by ecxs** [Budibase/budibase#9190] [Help]
Let me start off by saying that I really appreciate what the guys at BudiBase are doing, as well that nothing is perfect ... except myself maybe 😉

Also, let me explain where I’m coming from, and what I’m trying to achieve. I’m a FileMaker developer since the early 90s, and around the millennium, I started building an ERP system, that is now in use with a few small to medium size, high tec manufacturing companies, mainly in The Netherlands and Belgium. Since a couple of years, I’m also built quite a few (what I call) web portals for my customer’s customers and suppliers. For this, I used FileMaker’s WebDirect, as well as AppDrag. Recently, I discovered BudiBase, which now is my favorite tool for developing data centric web apps. If you want, you can read more about this here ... https://www.ecxs.com/blog/13-BudiBase-web-portals.html

Currently, I’m looking at building an integration between FileMaker and BudiBase. I’ve already built a tool that can publish FileMaker data via MySQL, and can keep data in sync between both tables. The only thing that’s currently missing, is attachments. And since the majority of my portals needs some kind pictures and/or documents, I’m a little stuck here. I’ve been looking at the following scenarios, which I would like to share with you. Perhaps somebody else has tried something similar, or has a better solution.

Scenario 1: blob support. Would be really nice, but not on the roadmap yet ... https://github.com/Budibase/budibase/projects/26

Scenario 2: file stored in public web folder via url. I’ve managed to get this working with a GitHub private repository with Pages. This works fine sharing files from FileMaker to BudiBase, but I have not found a solution to upload files from BudiBase to GitHub yet. And although I can create files with UUID’s, this can be a security issue, since it’s only security by obscurity.

Scenario 3: storing a file as base64 in a json object inside a text field. Files can be uploaded with the help of the Budibase File Upload Tool. The only downside, is that this tool does not support drag and drop.
https://github.com/Budibase/budibase/discussions/9160

Scenario 4: file uploaded directly to BudiBase attachment field/row. This is unfortunately not supported via the public API. I’ve managed uploading files to the prod-budi-app-assets bucket in MinIO, but have not yet found a method to upload it to the underlying app/assets bucket.
https://github.com/Budibase/budibase/discussions/9168

So, long story short; I have not found a (perfect) solution yet, but in the meantime, I got to know and appreciate  BudiBase’s internal structure better ...
